### PR TITLE
[GLUTEN-3559][VL] Fix TPCDS Plan Suite

### PR DESCRIPTION
**Pull request by ayushi-agarwal**
## What changes were proposed in this pull request?
As part of this change https://github.com/apache/spark/pull/38151 in spark 3.4 a new config was introduced spark.sql.readSideCharPadding which adds extra expression in project list expression. Due to this Suite fails with this error:

**Unable to convert the expression staticinvoke(class org.apache.spark.sql.catalyst.util.CharVarcharCodegenUtils, StringType, readSidePadding, sm_type#106, 30, true, false, true)**
**java.lang.UnsupportedOperationException: Unable to convert the expression staticinvoke(class org.apache.spark.sql.catalyst.util.CharVarcharCodegenUtils, StringType, readSidePadding, sm_type#106, 30, true, false, true)
	at io.substrait.spark.expression.ToSubstraitExpression.default(ToSubstraitExpression.scala:46)
	at io.substrait.spark.expression.ToSubstraitExpression.translateUp(ToSubstraitExpression.scala:145)**
	at io.substrait.spark.expression.ToSubstraitExpression.translateUp(ToSubstraitExpression.scala:136)
	at io.substrait.spark.expression.ToSubstraitExpression.convert(ToSubstraitExpression.scala:57)
	at io.substrait.spark.expression.ToSubstraitExpression.apply(ToSubstraitExpression.scala:50)
	at io.substrait.spark.logical.ToSubstraitRel.toExpression(ToSubstraitRel.scala:260)
	at io.substrait.spark.logical.ToSubstraitRel.$anonfun$visitProject$1(ToSubstraitRel.scala:230)
	at scala.collection.immutable.List.map(List.scala:297)
	at io.substrait.spark.logical.ToSubstraitRel.visitProject(ToSubstraitRel.scala:230)
	at io.substrait.spark.logical.ToSubstraitRel.visitProject(ToSubstraitRel.scala:45)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlanVisitor.visit(LogicalPlanVisitor.scala:38)
	at org.apache.spark.sql.catalyst.plans.logical.LogicalPlanVisitor.visit$(LogicalPlanVisitor.scala:25)

To fix the suite I have disabled this config in suite.

(Fixes: \#3559)

## How was this patch tested?

Ran this in local

### CODE REVIEW

**Comment by ayushi-agarwal** [comment]
@JkSelf Please review

**Review by JkSelf — APPROVED** [review]
@ayushi-agarwal LGTM. Can you help to file a new issue to record this exception with enabling `SQLConf.READ_SIDE_CHAR_PADDING` parameter?

**Comment by ayushi-agarwal** [comment]
> @ayushi-agarwal LGTM. Can you help to file a new issue to record this exception with enabling `SQLConf.READ_SIDE_CHAR_PADDING` parameter?

Thanks @JkSelf I have opened this issue https://github.com/oap-project/gluten/issues/4294

File: substrait/substrait-spark/src/test/scala/io/substrait/spark/TPCDSPlan.scala [hunk 1]
```diff
@@ -27,6 +27,7 @@ class TPCDSPlan extends TPCDSBase with SubstraitPlanTestBase {
     sparkContext.setLogLevel("WARN")
 
     conf.setConf(SQLConf.DYNAMIC_PARTITION_PRUNING_ENABLED, false)
+    conf.setConf(SQLConf.READ_SIDE_CHAR_PADDING, false)
```
Comment (ulysses-you) [inline substrait/substrait-spark/src/test/scala/io/substrait/spark/TPCDSPlan.scala:30]: The CI did not pass... this config is created since Spark3.4

Comment (JkSelf) [reply substrait/substrait-spark/src/test/scala/io/substrait/spark/TPCDSPlan.scala:30]: It's strange that all the CI builds passed when it was merged. I have reverted this PR. Thanks.

Comment (ayushi-agarwal) [reply substrait/substrait-spark/src/test/scala/io/substrait/spark/TPCDSPlan.scala:30]: @JkSelf Only 5 checks run in the CI for this PR, any clue why others didn't run?

Comment (JkSelf) [reply substrait/substrait-spark/src/test/scala/io/substrait/spark/TPCDSPlan.scala:30]: @ayushi-agarwal Can you help to re-open and try again? Thanks.

**Pull request merged**